jablib/src/test/java/org/jabref/logic/externalfiles/LinkedFileHandlerTest.java

```diff
@@ -59,37 +62,53 @@ void renameFile(String expectedFileName, String originalFileName, String newFile
         assertEquals(expectedFileName, result);
     }
 
-    @ParameterizedTest(name = "{1} with {2} should be {0}")
+    @ParameterizedTest(name = "{1} with {2} should be {0} with citation key 'asdf'")
     @CsvSource({
             "asdf.pdf, '', pdf",
+            "asdf.pdf, https://example.com/file.pdf, pdf",
+            "asdf.pdf, https://example.com/file.pdf?query=test, pdf",
+            "asdf.pdf, https://example.com/file.doc, pdf",
+            "asdf.pdf, https://example.com/file, pdf",
+            "asdf.pdf, https://example.com/file.pdf, ''",
+            "asdf.pdf, https://example.com/, pdf",
+            "asdf.pdf, path/to/file.pdf, pdf",
+            "asdf.pdf, https://www.cncf.io/wp-content/uploads/2020/08/OAM-Webinar-V2.pdf, pdf"
+    })
+    void getSuggestedFileName(String expectedFileName, String link, String extension) {
+        when(filePreferences.getFileNamePattern()).thenReturn("[bibtexkey]");
+
+        final LinkedFile linkedFile = new LinkedFile("", link, "");
+        final LinkedFileHandler linkedFileHandler = new LinkedFileHandler(linkedFile, entry, databaseContext, filePreferences);
+
+        final String result = linkedFileHandler.getSuggestedFileName(extension.isEmpty() ? Optional.empty() : Optional.of(extension));
+        assertEquals(expectedFileName, result);
+    }
+
+    @ParameterizedTest(name = "{1} with {2} should be {0} with empty citation key")
+    @CsvSource({
+            "file.pdf, '', pdf",
             "file.pdf, https://example.com/file.pdf, pdf",
             "file.pdf, https://example.com/file.pdf?query=test, pdf",
             "file.pdf, https://example.com/file.doc, pdf",
             "file.pdf, https://example.com/file, pdf",
             "file.pdf, https://example.com/file.pdf, ''",
             "file.pdf, https://example.com/, pdf",
             "file.pdf, path/to/file.pdf, pdf",
+            "other.pdf, https://example.com/other.pdf, pdf",
+            "other.pdf, https://example.com/other.pdf?query=test, pdf",
+            "other.pdf, https://example.com/other.doc, pdf",
+            "other.pdf, https://example.com/other, pdf",
+            "other.pdf, https://example.com/other.pdf, ''",
+            "other.pdf, path/to/other.pdf, pdf",
             "OAM-Webinar-V2.pdf, https://www.cncf.io/wp-content/uploads/2020/08/OAM-Webinar-V2.pdf, pdf"
     })
-    void getSuggestedFileName(String expectedFileName, String link, String extension) {
-        if (link.isEmpty()) {
-            when(filePreferences.getFileNamePattern()).thenReturn("[bibtexkey]");
-        } else {
-            when(filePreferences.getFileNamePattern()).thenReturn("[bibtexkey] - [title]");
```

Comment:
This is lost. I don't know the intention written here, but there is a test of an empty link - and then `[title]` is expected -- maybe, this is very strange and can really be removed, but we need to think :)